content/React Query and management of server state.md

```diff
@@ -0,0 +1,75 @@
+# React Query and management of server state
+
+Some days ago the [React Summit Remote Edition](https://remote.reactsummit.com/) gathered tons of people behind the screen with a promise of put together popular speakers around the React Community like [Kent C. Dodds](https://kentcdodds.com/) and [Guillermo Rauch](https://rauchg.com/) among others, but one talk especially caught my attention, *React Query: It’s Time to Break up with your "Global State”!* by [Tanner Linsley](https://twitter.com/tannerlinsley) were joining two pain points that I had frequently with React, such as how to handle asynchronous information in the global state and how to restructure my code to be less repetitive and not force async data to follow synchronous flow.
+In this article, I explain more in deep which is the problem with the global state and how React Query solves it in an efficient-scalable way. 
+
+## The problem with global state
+Libraries like Redux, MobX and even Context API provided by React are based in the Global State pattern, which means that to avoid prop drilling between the components and have one single point to read and write from the components, the entire application exposes a global variable called _global state_.
+
+This pattern has many advantages as I mentioned before, but the problem **lies in the different types of data that potentially have the global state in our applications**.
+These data can be internal information about our application, like for example if the navbar is open or not, or information that has ownership with another source, for instance, the user information like his name, age, etc, that is provided by backend and can be modified in the frontend.
+
+These two types of data can be classified as a **client state and server state**.
+The main difference between each other is the server state has two sources of truth, the backend and the frontend. This causes that the server state has to synchronize all the time to prevent that his information is not out of date. As we know, is not performant at all and is not desired by our users(image put a spinner every time you do a request) request information all the time, for that reason **we need in some way show the old information and at the same time updating with the new information on the background every so often**. 
```

Comment:
`This causes that the server state has to synchronize all the time to prevent that _its_ information is not out of date.` `Its` because you are talking about the server. 

`As we know, _this_ (?) is not performant at all...`

`...is not performant at all and is not desired by our users(image put a spinner every time you do a request) request information all the time,` This part is a little confusing, I guess the point is that `this is not performant at all and it wouldn't be a good practice to request information all the time (imagine puttting a spinner every time you do a request)` is this close to what you tried to say?

Reply:
`, for that reason **we need some way to show the old information and at the same time update it with the new information on the background every so often**. `

Reply:
I see that you were separating paragraphs with a space, is there a reason you didn't do that for these last paragraphs?

Reply:
Yes totally, I want to express that the UX of the user it will be affected if we don't take care in consideration this practice, so your suggestion reflects well what I try to explain